cups/http-addrlist.c

```diff
@@ -234,8 +236,9 @@ httpAddrConnect2(
       if (fds[nfds] > max_fd)
 	max_fd = fds[nfds];
 #endif /* !HAVE_POLL */
-
+#ifdef O_NONBLOCK
```

Comment:
Addresses still need to be set, so this conditional stuff isn't correct.

Reply:
It is not read though, so what is the point?

Reply:
addrs is not read or even used

Reply:
Something here is definitely not working as intended if this is the case

Reply:
Only the code under ifdef O_block is reading from the addrlist array, so if not assigning to that variable is not correct, neither is the current code.